plugins/lookup/merge_variables.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Thales Netherlands
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author:
+      - Roy Lenferink (@rlenferink)
+      - Mark Ettema (@m-a-r-k-e)
+    name: merge_variables
+    short_description: merge variables with a certain suffix
+    description:
+        - This lookup returns the merged result of all variables in scope that matches the given suffix optionally
+          starting with an initial value.
+    version_added: 6.1.0
+    options:
+      _terms:
+        description:
+          - The suffix of the variable name (all available variables will be checked with endswith(suffix))
+        required: true
+      initial_value:
+        description:
+          - An initial value to start with
+        required: false
+      override_warning:
+        description:
+          - Print a warning when a key will be overwritten
+        default: false
+      override_error:
```

Comment:
Better have a `override` option that can take a choice of `ignore`, `warn`, `error`, instead of having two different boolean toggles to control this behavior.

Also I would set the default to `error` and add a warning that setting it to anything else will result in undefined behavior since the merge order can change between different playbook runs.

Reply:
Thanks for the feedback, I combined the options and also am giving a warning in case another option than `error` is chosen.